control-plane-operator/controllers/hostedcontrolplane/v2/kas/component.go

```diff
@@ -100,7 +102,19 @@ func NewComponent() component.ControlPlaneComponent {
 			component.EnableForPlatform(hyperv1.AWSPlatform),
 			component.WithAdaptFunction(adaptAWSPodIdentityWebhookKubeconfigSecret),
 		).
+		WithManifestAdapter(
+			"azure-kms-secretprovider.yaml",
+			component.WithAdaptFunction(kms.AdaptAzureSecretProvider),
+			component.WithPredicate(isAroHCP),
+		).
 		RolloutOnConfigMapChange("kas-config", "kas-audit-config", "auth-config").
 		RolloutOnSecretChange("kas-secret-encryption-config").
 		Build()
 }
+
+func isAroHCP(cpContext component.WorkloadContext) bool {
+	if cpContext.HCP.Spec.SecretEncryption != nil && cpContext.HCP.Spec.SecretEncryption.KMS != nil {
```

Comment:
I don' think `Spec.SecretEncryption.KMS != nil` is enough. We should check that `.Spec.SecretEncryption.Type == KMS`

nit: predicate function name now should be different. maybe just `enableAzureKMSSecretProvider`